src/diffusers/video_processor.py

```diff
@@ -50,94 +51,53 @@ def tensor2vid(
 
         return outputs
 
-    def preprocess_video(self, video) -> torch.FloatTensor:
+    def preprocess_video(self, video) -> torch.Tensor:
         r"""
         Preprocesses input video(s).
 
         Args:
             video: The input video. It can be one of the following:
                 * List of the PIL images.
                 * List of list of PIL images.
+                * 4D Torch tensors (expected shape for each tensor: (num_frames, num_channels, height, width)).
+                * 4D NumPy arrays (expected shape for each array: (num_frames, height, width, num_channels)).
                 * List of 4D Torch tensors (expected shape for each tensor: (num_frames, num_channels, height, width)).
-                * List of list of 4D Torch tensors (expected shape for tensor: (num_frames, num_channels, height,
-                  width)).
                 * List of 4D NumPy arrays (expected shape for each array: (num_frames, height, width, num_channels)).
-                * List of list of 4D NumPy arrays (expected shape for each array: (num_frames, height, width,
-                  num_channels)).
-                * List of 5D NumPy arrays (expected shape for each array: (batch_size, num_frames, height, width,
-                  num_channels).
-                * List of 5D Torch tensors (expected shape for each array: (batch_size, num_frames, num_channels,
-                  height, width).
                 * 5D NumPy arrays: expected shape for each array: (batch_size, num_frames, height, width,
                   num_channels).
                 * 5D Torch tensors: expected shape for each array: (batch_size, num_frames, num_channels, height,
                   width).
         """
-        supported_formats = (np.ndarray, torch.Tensor, PIL.Image.Image, list)
-
-        # Single-frame video.
-        if isinstance(video, supported_formats[:-1]):
-            video = [video]
-
-        # List of PIL images.
-        elif isinstance(video, list) and isinstance(video[0], PIL.Image.Image):
+        if isinstance(video, list) and isinstance(video[0], np.ndarray) and video[0].ndim == 5:
+            warnings.warn(
+                "Passing `video` as a list of 5d np.ndarray is deprecated."
```

Comment:
Technically not deprecated since we didn't support it before right? I think we just raise an error here.

Reply:
it was kinda supported before https://github.com/huggingface/diffusers/blob/0d23645bd120e6785a3d81ee8e053afe84bbc77f/src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_synth_img2img.py#L119
this code, if you pass a list of 5d tensors, it would work; I think it is because of the way the code was written, here https://github.com/huggingface/diffusers/blob/0d23645bd120e6785a3d81ee8e053afe84bbc77f/src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_synth_img2img.py#L123 the first thing it does is to check if it is a tensor, and if so add it to a list; so in order to support a 5d tensor, it has to support a list of 5d tensor as well. This is same situation with image processor as well